scripts/seastar-addr2line

```diff
@@ -108,6 +108,15 @@ cmdline_parser.add_argument(
         ' The amount of non-backtrace lines considered can be set with --before.'
         ' By default no matching is performed.')
 
+cmdline_parser.add_argument(
+        '-a',
+        '--addr2line',
+        type=str,
+        metavar='CMD_PATH',
+        default='addr2line',
+        help='The path or name of the addr2line command, which should behave as and '
+        'accept the same options as binutils addr2line or llvm-addr2line.')
+
```

Comment:
The parameter is fine, but good default selection is more important. Users shouldn't have to remember to select a different addr2line if seastar-addr2line knows there is a better alternative.

Reply:
Right, I had considered also adding auto-detection if `llvm-addr2line` was on the PATH, though I saw a few downsides.

I could certainly add that in a new commit in this series, though most likely almost nobody has it on their path since you have to install llvm package (not clang) to get it AFAICT.

Reply:
> though I saw a few downsides.

curious what these were, especially if llvm-addr2line is strictly better suited and in PATH upon inspect.

Reply:
Funny, I had a longer message but then I thought "Avi seems like likes short and to the point messages" so I fought my natural tendency to over-explain and truncated it, but here they are:

 - I am not sure that `llvm-addr2line` is equivalent to `addr2line`, as it is using a totally different ELF/DWARF decoding engine, etc. I did find at least one difference with the way empty lines are handled (which required a fix in seastar-addr2line) but there could be others lurking so I thought it was more conservative to make it an optional non-default thing for the initial change, let it bake for a while.
 - In our case, I want to point to the llvm-a2l binary in our own built clang directory, so in any case a path option seemed needed.
 - Installing LLVM seems to take about 500 MB of extra disk space (on my Ubuntu box) which may be annoying e.g., for a Dockerfile which sets up the decoding environment.
 - At least on Ubuntu, installing llvm doesn't actually install `llvm-addr2line`, but rather `llvm-addr2line-13` (presumably the version number), so now I guess I would have to autodetect that (e.g., by stepping through some candidate versions?) or ask people to symlink the bare name into their path.
 
FWIW, once you are symlinking `llvm-addr2line` to `llvm-addr2line-13` you might as well just symlink it to `addr2line` (in which case most of this change arguably isn't needed at all: only the part to handle the slightly different empty line behavior is needed).

All that said, I'm fine with adding autodetect too.

Reply:
I don't mind long messages, as long as they have a good number of footnotes.